File: hejoo345/week_21/pgs_입국심사.js
function solution(n, times) {
    times.sort((a, b) => a - b);
    let left = 1;
    let right = n * times[times.length - 1];
    let answer = right;

    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        let cnt = 0;
        times.forEach(time => {
            cnt += Math.floor(mid / time);
            if (cnt >= n) {
                answer = Math.min(mid, answer);
                return;
            }
        });
        if (cnt >= n) {
            right = mid - 1;
        } else {
            left = mid + 1;
        }
    }
    return answer;
}

File: OneK-2/week_21/PGS_입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        Arrays.sort(times);

        long left = 1;
        long right = (long) times[times.length - 1] * n;
        long mid = 0;
        long sum = 0;

        while (left <= right) {
            sum = 0;
            mid = (left + right) / 2;

            for (int t : times) {
                sum += mid / t;
            }

            if (sum >= n) {
                answer = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return answer;
    }
}